frontend/src/entities/configuration/configuration-reducer.ts

```diff
@@ -0,0 +1,142 @@
+/**
+  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+  Licensed under the Apache License, Version 2.0 (the "License").
+  You may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+import {
+    createAsyncThunk,
+    createSlice,
+    isFulfilled,
+    isPending,
+    PayloadAction,
+} from '@reduxjs/toolkit';
+import axios from '../../shared/util/axios-utils';
+import { defaultConfiguration, IAppConfiguration } from '../../shared/model/app.configuration.model';
+
+const initialState = {
+    loadingAppConfigUpdate: false,
+    loadingAppConfigList: false,
+    loadingAppConfig: false,
+    appConfigList: [],
+    appConfig: defaultConfiguration,
+};
+
+// Actions
+export const getConfiguration = createAsyncThunk(
+    'app_config/get_configuration',
+    async (props: GetAppConfigurationProps) => {
+        const searchParams = new URLSearchParams();
+        searchParams.set('configScope', props.configScope);
+        const requestUrl = `/app-config?${searchParams.toString()}`;
```

Comment:
FWIW if you're using a variable inside `${...}` it implicitly has toString() called and you don't need to call it explicitly.